### ISSUE TITLE
Alter the behaviour for specific struct fields in a convenient way

### ISSUE THREAD

**Issue by r4gus** [r4gus/zbor#13]
Currently, we use 'decorators for poor people' (e.g., `#x`) to express a specific intent. There are some proposals when it comes to Zig decorators but I've found another solution: https://github.com/ziglang/zig/pull/8987#issuecomment-856623134.

Would look something like this:

```zig
test "field alias escapes" {
    const E = enum(u8) { v1 = 1, v2 = 2 };
    const S = struct { foo: E };
    const s = S{ .foo = E.v2 };

    const s = try cbor.stringify(writer, s, .{
        .field_settings = &.{.{ .field = "foo", .alias =  "__foo" , .options = . { ... }}},
        .allocator = std.testing.allocator,
    });
}
```

**Comment by r4gus** [comment]
allow the renaming of struct fields, e.g., `alg` => `1`

**Comment by r4gus** [comment]
Solved in branch 0.11.2-alpha. Issue will be closed when 0.11.2-alpha is merged into master
